### PR TITLE
chore(examples): update vote-helm example to use action configs

### PR DESCRIPTION
**Pull request by Walther**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes https://github.com/garden-io/garden/issues/3767

**Special notes for your reviewer**:

- `garden validate` passes
- `garden deploy` passes
- `garden run db-init` and `garden run db-clear` pass
- `garden cleanup env` passes

Squash before merging.

### CODE REVIEW

File: core/src/actions/base.ts [hunk 1]
```diff
@@ -89,7 +89,7 @@ const actionSourceSpecSchema = () =>
     .xor("path", "repository")
     .meta({ name: "action-source", advanced: true, templateContext: ActionConfigContext })
 
-export const includeExcludeSchema = () => joi.array().items(joi.posixPath().allowGlobs().subPathOnly())
+export const includeExcludeSchema = () => joi.array().items(joi.posixPath().allowGlobs())
```
Comment (Walther) [inline core/src/actions/base.ts:92]: ⚠️  Note this change here

File: examples/vote-helm/api/garden.yml [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
-    tag: ${modules.api-image.version}
-  ingress:
-    enabled: true
-    paths: [/]
-    hosts: ["api.${var.baseHostname}"]
-  healthCheckPath: /api
+  - deploy.redis
+spec:
+  chart:
+    path: ../base-chart/
```
Comment (Walther) [inline examples/vote-helm/api/garden.yml:9]: ⚠️  Note the change required in order to be able to use relative path here

File: examples/vote-helm/result/garden.yml [hunk 3]
```diff
@@ -1,23 +1,18 @@
...
-tests:
-  - name: integ
-    args: [echo, ok]
-    dependencies:
-      - db-init
\ No newline at end of file
+  - run.db-init
+spec:
+  chart:
+    path: ../base-chart/
```
Comment (Walther) [inline examples/vote-helm/result/garden.yml:9]: ⚠️  and here

File: core/src/actions/base.ts (hunk 1 above)
Comment (edvald) [reply core/src/actions/base.ts:92]: We can't actually make this change, sorry. This won't play nice with version calculation and build staging. It is also a conscious design choice not to allow pulling files from outside the module root.

**Review by vvagaytsev — APPROVED** [review]

**Pull request merged**